### PR TITLE
Fix/enhance stubgen docstrings for functions, properties and types

### LINKED ISSUE #16114: The "include-docstrings" option doesn't work
After cloning the latest Mypy version available on GitHub, I see the "include-docstrings" option available, but running stubgen on an extension module that provides docstrings results in no docstrings at all.

**To Reproduce**

```git
git clone https://github.com/python/mypy.git
```

```python
python -m build
python -m pip install mypy-1.7.0+dev-py3-none-any.whl
```

```shell
stubgen --include-docstrings -p myextmodule[.pyd]
```

```python
python -m mypy.stubgen --include-docstrings -p myextmodule[.pyd]
```

**Expected Behavior**

The .pyi file generated from stubgen should contain docstrings declared in the extension module, for classes and functions.

**Actual Behavior**

There is no docstrings at all, but they exist in the extension module. I have confirmed using help(myextmodule).

**Your Environment**

python 3.11 for Windows x64
myextmodule installed onto site-packages

- Mypy version used: 1.7.0+dev
- Mypy command-line flags: --include-docstrings -p myextmodule
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: cp3.11

I know this is a new feature, but it seems not to work out.

### PR DESCRIPTION
**Pull request by lmbelo**
Fixes #16114

Outputs docstrings for function, properties and types provided by extension modules to the stub file using stubgen.
Note: functions with overloaded signatures are not stripped.

### CODE REVIEW

File: mypy/stubgenc.py [hunk 1]
```diff
@@ -476,25 +478,39 @@ def infer_prop_type(docstr: str | None) -> str | None:
...
+        # Quoted docstr
+        docstr = mypy.util.quote_docstring(docstr.strip())
+        # Indented docstr
+        docstr = "\n    ".join(docstr.split("\n"))
+
     if is_static_property(obj):
         trailing_comment = "  # read-only" if readonly else ""
         static_properties.append(f"{name}: ClassVar[{inferred}] = ...{trailing_comment}")
+        if include_docstrings and docstr:
+            static_properties.extend(f"{docstr}".split("\n"))
```
Comment (ikonst) [inline mypy/stubgenc.py:503]: Why format a string?

File: mypy/stubgenc.py [hunk 2]
```diff
@@ -476,25 +478,39 @@ def infer_prop_type(docstr: str | None) -> str | None:
...
     else:  # regular property
         if readonly:
             ro_properties.append("@property")
             ro_properties.append(f"def {name}(self) -> {inferred}: ...")
+            if include_docstrings and docstr:
+                ro_properties.extend(f"    {docstr}".split("\n"))
```
Comment (ikonst) [inline mypy/stubgenc.py:509]: Did you mean to indent _after_ splitting?

**Review by ikonst — COMMENTED** [review]
Does it necessitate changing tests?

File: mypy/stubgenc.py (hunk 1 above)
Comment (lmbelo) [reply mypy/stubgenc.py:503]: Oh, my mistake. I will replace unnecessary formats.

File: mypy/stubgenc.py (hunk 2 above)
Comment (lmbelo) [reply mypy/stubgenc.py:509]: Yes, right.

`
def my_func(self):
    '''docstrings here'''
`

**Comment by wrinch** [comment]
Any chance this can get merged in?  I'd really like to test it out.

**Review by JelleZijlstra — COMMENTED** [review]
Can you put in a guard against non-str docstrings? I know it shouldn't happen but it's possible at runtime, and I'd rather not have stubgen crash on any code.

```
>>> def f(): pass
... 
>>> f.__doc__ = 42
>>> f.__doc__
42
```

No need to handle them specially, just ignore all docstrings that aren't instances of str.

**Comment by lmbelo** [comment]
> Can you put in a guard against non-str docstrings? I know it shouldn't happen but it's possible at runtime, and I'd rather not have stubgen crash on any code.
> 
> ```
> >>> def f(): pass
> ... 
> >>> f.__doc__ = 42
> >>> f.__doc__
> 42
> ```
> 
> No need to handle them specially, just ignore all docstrings that aren't instances of str.

Yes, sure. I Will do.

**Pull request closed without merging**